core/src/main/java/bisq/core/CoreModule.java

```diff
@@ -96,6 +98,7 @@ protected void configure() {
         install(alertModule());
         install(filterModule());
         install(corePresentationModule());
+        bind(PubKeyRing.class).toProvider(PubKeyRingProvider.class);
```

Comment:
Is this guaranteed to be in singleton context?
Can you move it to the other bindings above, so th einstall calls are not mixed with bind calls?

Reply:
Ah from the test below its seems its singleton.

Reply:
yes i made sure that it is.